server/app/assets/javascripts/admin_application_view.ts

```diff
@@ -0,0 +1,66 @@
+class AdminApplicationView {
+  private static APPLICATION_STATUS_SELECTOR =
+    '.cf-program-admin-status-selector'
+  private static CONFIRM_MODAL_FOR_DATA_ATTRIBUTE =
+    'data-status-update-confirm-for-status'
+
+  constructor() {
+    this.registerStatusSelectorEventListener()
+  }
+
+  private registerStatusSelectorEventListener() {
+    const statusSelectForm = document.querySelector(
+      AdminApplicationView.APPLICATION_STATUS_SELECTOR,
+    ) as HTMLFormElement | null
+    if (!statusSelectForm) {
+      // If status tracking isn't enabled, there's nothing to do.
+      return
+    }
+    const statusSelector = this._assertNotNull(
+      statusSelectForm.querySelector('select') as HTMLSelectElement | null,
+      'status selector',
+    )
+
+    // The original value is tracked since neither the 'change/input' events provide the previous
+    // value prior to input.
+    const originalSelectedValue = statusSelector.value
+    statusSelector.addEventListener('change', (event) => {
+      // The promise result is not awaited since the modal is responsible for updating the status
+      // and we only care about showing it.
+      this.showConfirmStatusChangeModal(statusSelector.value)
+      // Reset the selection to its previous value in the case that the user cancels the
+      // confirmation dialog.
+      statusSelector.value = originalSelectedValue
```

Comment:
Ah ok, I think it could still be confusing, but not imp right now. Can ask Raha too

EG what if I'm changing from `in-progress` to `approved`, and the background still says `in-progress` even though I just selected something different, I might not have confidence the system is going to do the right thing.

Reply:
Fair. The dialog will also show the original state as well.